SYCL/ESIMD/api/functional/ctors/ctor_default.cpp

```diff
@@ -0,0 +1,167 @@
+//==------- ctor_default.cpp  - DPC++ ESIMD on-device test ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu || level_zero
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Test for esimd default constructor.
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <string>
+#include <sycl/ext/intel/experimental/esimd.hpp>
+
+#include "../../../esimd_test_utils.hpp"
+#include "../logger.hpp"
+#include "../type_coverage.hpp"
+
+using namespace cl::sycl;
+using namespace sycl::ext::intel::experimental;
+using namespace sycl::ext::intel::experimental::esimd;
+using namespace esimd_test::api::functional::type_coverage;
+using namespace esimd_test::api::functional::logger;
+
+// Dummy kernel for submitting some code into device side.
+template <typename DataTypeT, int TestNumber, int NumElems>
+struct kernel_for_simd;
+
+// Dummy kernel for submitting some code into device side.
+template <typename DataT, int NumElems, typename T> struct Kernel;
+
+// Struct that provide method that let get current test case description and
+// method that let call simd constructor for initializer test case
+struct initializer {
+  static std::string get_description() { return "initializer"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT *output_data) SYCL_ESIMD_KERNEL {
```

Comment:
That's interesting. Do we really need the `SYCL_ESIMD_KERNEL` for every function used in kernel? I assumed we need it only for the kernel itself..

Reply:
this comment was applied in this commit: c201b251096c6d476ae70c763abe09fd4cd801d8